Dokumenty/Datový model/readme_ermodel.txt

```diff
@@ -0,0 +1,25 @@
+Er model
+
+Entita Položka/Zboží 
+	- Název, Cena, skladem (bool)
+	- id_kat - vazba 1:N, tedy že jedno zboží může být ve vícero kategoriích (Třeba fitness hodinky budou patřit do hodinek ale i do elektroniky, popř. něco v tom smyslu)
+	- id_vyr - jedna položka muže mít pouze jednoho výrobce. Zboží také může mít víc výrobců, pak by byl vztah 1:N.
+	- id_bar - jeden výrobek se může prodávat ve vícerobarvách, sem bych dal nepovinný vztah, neboť vždy firma nemusí určit barvy výrobku
+	- id_stav - stav bych čísloval - 1,2,3,4,5 ... 1 - Nový, 2 - Rozbalený, 3 - Použitý
+			  - teoreticky stav by šel nahradit kategorií s názvem Rozbalené výrobky, použité apod.
+			  
+			  
+Entita Kategorie
+	- ID, Název
+	- Id_Podkategorie - myšleno tak, že jedna kategorie může patřit do jiné kategorie, vazba N:N
+	
+- Entita štítek
+	- Id, Název
+	- Cizí klíč Štítek - Hodnota_Štítku, tj. jeden štítek může obsahovat vícero hodnot, dám třeba štítek záruka a hodnoty budou: 12,24,48, ...., povinná vazba
+	- pomocí štítku mužu oštítkovat kategorie celé a nebo položky
+	
+- Entita kategorie, výrobce, barva, Stav - atributy unique a not null
+	
```

Comment:
podle diagramu v souboru `er model.png` to nemůže fungovat správně, protože `Položka` má vazbu do `vazebni_tab` a ta má vazbu na `štítek`. Tím jsme schopní tedy zajistit, že `Produkt id 123` má vazbu na `štítek id 11 záruka`, ale kde se uchová informace, že `Produkt id 123` má záruku `2 roky`? což není štítek, ale `Hodnota_Stitek`.
Tzn. položka musí mít primárně vazbu na `Hodnota_Stitek` místo `štítek`

Reply:
Ano, tak to je chyba, to jsem si neuvědomil. Jedině mít štítek s názvem ,,záruka - 2 roky" s hodnotou 2 a ty produkty oštítkovat tímto štítkem. Což uznávám, že není optimální toto řešení.

Reply:
Ale třeba s těmi hodinkami s vodotěsností nebo ciferník s různými hodnotami, tak to bylo funkční.